extract/toc.js

```diff
@@ -66,6 +66,27 @@ export default function toc(tocHtml, manifest, spine) {
   };
 }
 
+function findTocRoot(currentNode, found = {}) {
+  Object.keys(currentNode).forEach(key => {
```

Comment:
This method needs a bit of a rework. Overwriting params is generally not a good idea (mainly because of us remembering not to screw up with objects passed by reference, etc.). I'd say it's a good idea to extract that out. Also, your iteration is breaking from the loop, so you should be using `find` instead